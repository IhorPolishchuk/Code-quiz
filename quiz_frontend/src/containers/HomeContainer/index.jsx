import React, { Component } from 'react';

import { TextField } from'./../../components/ui';

class HomeContainer extends Component {
    render () {
        return (
            <div>
                <TextField
                    label="text"
                    error={false}
                    regex={/\.com/g}
                    errorMessage="Should contain '.com'"
                />
            </div>
        );
    }
}

export default HomeContainer;
