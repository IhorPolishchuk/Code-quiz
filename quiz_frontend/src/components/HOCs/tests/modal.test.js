import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Modal } from './../index.js';
import HomeContainer from './../../../containers/HomeContainer';

const dummy = {
    title: 'Modal'
};

describe('Modal HOC test', () => {
    it('Modal HOC without children', () => {
        const modal = shallow(
            <Modal
                title={dummy.title}
            />
        );
        const tree = toJson(modal);
        expect(modal.contains(<div className="modal__content"/>)).toBeTruthy();
        expect(modal.instance().props.title).toBe(dummy.title);
        expect(tree).toMatchSnapshot();
    });
    it('Modal HOC with children', () => {
        const modal = shallow(
            <Modal
                title={dummy.title}
            >
                <HomeContainer/>
            </Modal>
        );
        const tree = toJson(modal);
        
        expect(modal.instance().props.title).toBe(dummy.title);
        expect(modal.contains(<div className="modal__content"/>)).toBeFalsy();
        expect(modal.find(<HomeContainer/>)).toBeTruthy();
        expect(tree).toMatchSnapshot();
    });
});
