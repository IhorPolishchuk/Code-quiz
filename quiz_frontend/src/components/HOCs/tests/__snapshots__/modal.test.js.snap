// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal HOC test Modal HOC with children 1`] = `
<div
  className="modal__wrapper"
>
  <div
    className="modal__body"
  >
    <div
      className="modal_top__wrapper"
    >
      <div
        className="modal__title"
      >
        Modal
      </div>
      <Button
        className="modal__close modal__button"
        icon="M672 256l-160 160-160-160-96 96 160 160-160 160 96 96 160-160 160 160 96-96-160-160 160-160z"
        iconLeft={true}
        iconRight={false}
        title=""
      />
    </div>
    <div
      className="modal__content"
    >
      <HomeContainer />
    </div>
    <hr />
    <div
      className="modal_bottom__wrapper"
    >
      <Button
        className="modal__button"
        icon="M672 256l-160 160-160-160-96 96 160 160-160 160 96 96 160-160 160 160 96-96-160-160 160-160z"
        iconLeft={true}
        iconRight={false}
        title="Cancel"
      />
      <Button
        className="modal__button"
        icon="M864 128l-480 480-224-224-160 160 384 384 640-640z"
        iconLeft={true}
        iconRight={false}
        title="Сonfirm"
      />
    </div>
  </div>
</div>
`;

exports[`Modal HOC test Modal HOC without children 1`] = `
<div
  className="modal__wrapper"
>
  <div
    className="modal__body"
  >
    <div
      className="modal_top__wrapper"
    >
      <div
        className="modal__title"
      >
        Modal
      </div>
      <Button
        className="modal__close modal__button"
        icon="M672 256l-160 160-160-160-96 96 160 160-160 160 96 96 160-160 160 160 96-96-160-160 160-160z"
        iconLeft={true}
        iconRight={false}
        title=""
      />
    </div>
    <div
      className="modal__content"
    />
    <hr />
    <div
      className="modal_bottom__wrapper"
    >
      <Button
        className="modal__button"
        icon="M672 256l-160 160-160-160-96 96 160 160-160 160 96 96 160-160 160 160 96-96-160-160 160-160z"
        iconLeft={true}
        iconRight={false}
        title="Cancel"
      />
      <Button
        className="modal__button"
        icon="M864 128l-480 480-224-224-160 160 384 384 640-640z"
        iconLeft={true}
        iconRight={false}
        title="Сonfirm"
      />
    </div>
  </div>
</div>
`;
